polkadot/roadmap/implementers-guide/src/protocol-validator-disabling.md

```diff
@@ -0,0 +1,281 @@
+
+</br></br></br>
+
+# Economic consequences of Disablement
+
+Disablement is generally a form of punishment and that will be reflected in the rewards at the end of an era. A disabled validator will not receive any rewards for backing or block authoring. which will reduce its profits.
+
+That means that the opportunity cost of being disabled is a punishment by itself and thus it can be used for some cases where a minor punishment is needed. Current implementation was using 0% slashes to mark nodes for chilling and similar approach of 0% slashes can be used to mark validators for disablement. ([**Point 1.**](#system-overview))
+0% slashes could for instance be used to punish approval checkers voting invalid on valid candidates.
+
+Anything higher than 0% will of course also lead to a disablement.
+
+> **Notes:** \
+> Alternative designs incorporating disabling proportional to offenses were explored but they were deemed too complex and not worth the effort. Main issue with those is that proportional disabling would cause back and forth between disabled and enabled which complicated tracking the state of disabled validators and messes with optimistic node optimizations. Main benefits were that minor slashes will be barely disabled which has nice properties against sacrifice attacks.
+
+</br></br></br>
+
+# Redundancy
+
+Some systems can be greatly simplified or outright removed thanks to the above changes. This leads to reduced complexity around the systems that were hard to reason about and were sources of multiple bugs.
+
+## Automatic Chilling
+
+Chilling is process of a validator dropping theirs intent to validate. This removes them from the upcoming NPoS solutions and effectively pushes them out of the validator set as quickly as of the next era (or 2 era in case of late offenses). All nominators of that validator were also getting unsubscribed from that validator. Validator could re-register their intent to validate at any time.
+
+Chilling had a myriad of problems. It assumes that validators and nominators remain very active and monitor everything. If a validator got slashed he was getting automatically chilled and his nominators were getting unsubscribed. This was an issue because of minor non-malicious slashes due to node operator mistakes or small bugs. Validators got those bugs fixed quickly and were reimbursed but nominator had to manually re-subscribe to the validator, which they often postponed for very lengthy amounts of time most likely due to simply not checking their stake. This forced unsubscribing of nominators was later removed but it leads back to the original quoted issue of offending validators simply re-registering their interest and continuing to attack the network. 
```

Comment:
> and his nominators were getting unsubscribed

as noted, this is not correct anymore. 

An underlying flaw that is being shown here is that, do we have any tests rn that forks polkadot and simulates a slash in it? 

I think both for staking devs (@Ank4n and @gpestana) and the parachain runtime team this is super important to have. 

We have built `pallet-root-offence` exactly with this idea in mind, but never exercised it AFAIK. It should be possible with chopsticks to run an altered Polkadot runtime that is the same + this pallet, then trigger a slash in the UI. 

Again, I think this is a super important scenario to have ready both as a manual test for monkey testing, and automated for integration testing.

Reply:
> as noted, this is not correct anymore.

Which is good. We assume this is not the case. Otherwise we would still be having the problem that determinism issues could be used to kick out honest nodes.

I agree about more tests. But I think what @Overkillus  is saying is not conflicting with what you are saying:

He is saying that it used to be like that (not any more), and you are saying the same - right?